controllers/result_test.go

```diff
@@ -33,16 +34,93 @@ var _ = Describe("ResultController", func() {
 	})
 
 	Describe("POST /results", func() {
-		Context("given user already signed in", func() {
+		Context("given a valid CSV file", func() {
 			It("redirects to root path", func() {
 				user := FabricateUser("dev@nimblehq.co", "password")
-				body := GenerateRequestBody(nil)
-				response := MakeAuthenticatedRequest("POST", "/results", body, user)
+				header, body := CreateRequestInfoFormFile("tests/fixtures/files/valid.csv")
+
+				response := MakeAuthenticatedRequest("POST", "/results",  header, body, user)
+				currentPath := GetCurrentPath(response)
+
+				Expect(response.StatusCode).To(Equal(http.StatusFound))
+				Expect(currentPath).To(Equal("/"))
+			})
+
+			It("sets the success message", func() {
+				user := FabricateUser("dev@nimblehq.co", "password")
+				header, body := CreateRequestInfoFormFile("tests/fixtures/files/valid.csv")
+
+				response := MakeAuthenticatedRequest("POST", "/results", header, body, user)
+				flash := GetFlashMessage(response.Cookies())
+
+				Expect(flash.Data["success"]).To(Equal("Successfully uploaded the file, the result status would be update soon"))
+				Expect(flash.Data["error"]).To(BeEmpty())
+			})
+
+			It("created results with given keywords", func() {
```

Comment:
How about starting with `It` with present tense verb? 
```suggestion
			It("creates results with given keywords", func() {
```

Reply:
Nice catch 🙌🏼  
fixed in [47d92bf](https://github.com/carryall/go-google-scraper-challenge/pull/56/commits/47d92bf0b0bfc660de2a408aba7a639c5e92ca7f)